mkosi

```diff
@@ -1228,6 +1255,58 @@ def patch_root_uuid(args, loopdev, root_hash):
         subprocess.run(["sfdisk", "--part-uuid", loopdev, str(args.root_partno), str(u)],
                        check=True)
 
+def install_unified_kernel(args, workspace, run_build_script, root_hash):
+
+    # Iterates through all kernel versions included in the image and
+    # generates a combined kernel+initrd+cmdline+osrelease EFI file
+    # from it and places it in the /EFI/Linux directory of the
+    # ESP. sd-boot iterates through them and shows them in the
+    # menu. These "unified" single-file images have the benefit that
+    # they can be signed like normal EFI binaries, and can encode
+    # everything necessary to boot a specific root device, including
+    # the root hash.
+
+    if not args.bootable:
+        return
+
+    if args.distribution != Distribution.fedora:
+        return
```

Comment:
Perhaps it would be cleaner to put everything below in an `install_unified_kernel_fedora` function and make `install_unified_kernel` delegate to the appropriate function for the distribution? Arch Linux, at least, will probably look quite different (use `mkinitcpio` instead of `dracut` (or `kernel-install`?) and then patch together the executable with `objcopy`).